connectors/spark/src/test/java/io/unitycatalog/spark/auth/AwsCredentialRenewalTest.java

```diff
@@ -0,0 +1,362 @@
+    serverProperties.put("s3.sessionToken.0", "sessionToken0");
+    serverProperties.put("s3.credentialsGenerator.0", CREDENTIALS_GENERATOR);
+  }
+
+  private SparkSession createSparkSession() {
+    String testCatalogKey = String.format("spark.sql.catalog.%s", CATALOG_NAME);
+    return SparkSession.builder()
+        .appName("test-credential-renewal")
+        .master("local[1]") // Make it single-threaded explicitly.
+        .config("spark.sql.extensions", "io.delta.sql.DeltaSparkSessionExtension")
+        .config(
+            "spark.sql.catalog.spark_catalog", "org.apache.spark.sql.delta.catalog.DeltaCatalog")
+        .config("spark.hadoop." + UCHadoopConf.UC_TEST_CLOCK_NAME, CLOCK_NAME)
+        .config("spark.hadoop." + UCHadoopConf.UC_RENEWAL_LEAD_TIME_KEY, 0L)
+        .config("spark.sql.shuffle.partitions", "1")
+        .config(testCatalogKey, UCSingleCatalog.class.getName())
+        .config(testCatalogKey + ".uri", serverConfig.getServerUrl())
+        .config(testCatalogKey + ".token", serverConfig.getAuthToken())
+        .config(testCatalogKey + ".warehouse", CATALOG_NAME)
+        .config(testCatalogKey + ".renewCredential.enabled", "true")
+        .config("fs.s3.impl", S3CredFileSystem.class.getName())
+        .getOrCreate();
+  }
+
+  private interface Callable {
+
+    void call() throws Exception;
+  }
+
+  private void callQuietly(Callable call) {
+    try {
+      call.call();
+    } catch (Exception e) {
+      // ignored.
+    }
+  }
+
+  @BeforeEach
+  public void beforeEach() throws Exception {
+    super.setUp();
+    session = createSparkSession();
+
+    // Initialize the catalog in unity catalog server.
+    catalogOperations.createCatalog(
+        new CreateCatalog().name(CATALOG_NAME).comment("Spark catalog"));
+
+    // Initialize the schema in unity catalog server.
+    schemaOperations = new SdkSchemaOperations(createApiClient(serverConfig));
+    schemaOperations.createSchema(new CreateSchema().name(SCHEMA_NAME).catalogName(CATALOG_NAME));
+  }
+
+  @AfterEach
+  public void afterEach() {
+    // Drop the table.
+    sql("DROP TABLE IF EXISTS %s", TABLE_NAME);
+
+    // Delete the scheme.
+    callQuietly(() -> schemaOperations.deleteSchema(SCHEMA_NAME, Optional.of(true)));
+
+    // Delete the catalog.
+    callQuietly(() -> catalogOperations.deleteCatalog(CATALOG_NAME, Optional.of(true)));
+
+    // Close the session.
+    callQuietly(() -> session.close());
+
+    super.cleanUp();
+  }
+
+  @AfterAll
+  public static void afterAll() {
+    Clock.removeManualClock(CLOCK_NAME);
+  }
+
+  public static Clock testClock() {
+    return Clock.getManualClock(CLOCK_NAME);
+  }
+
+  @Test
+  public void testFileSystemRenewal() throws Exception {
+    String location = String.format("s3://%s%s/fs", BUCKET_NAME, dataDir.getCanonicalPath());
+    Path locPath = new Path(location);
+
+    // Create the external delta path-based table.
+    S3CredFileSystem.credentialCheckEnabled = false;
```

Comment:
why do you need to disable this?

Reply:
and why do you need to create the path-based table separately and than again calling create to define the catalog table? what are you achieving with this two step instead of one step?

Reply:
Left an issue here: https://github.com/unitycatalog/unitycatalog/issues/1139

Reply:
Oh,  sorry.  we currently only not support the path-based table creation without specified location. 

The SQL `CREATE TABLE %s (id INT) USING delta LOCATION '%s'` should be fine. Thanks for the reminder.